### PR TITLE
[SCFD-3148] Added private_attribute_id for all entities 

### PR DESCRIPTION
**Pull request by benflexcompute**
to ensure that front end entity tracking still works

### CODE REVIEW

**Comment by benflexcompute** [comment]
Pending testing with Tengda. Labeled WIP

File: examples/tutorials/quick_start.py [hunk 1]
```diff
@@ -20,13 +20,38 @@
 fl.Env.preprod.active()
 
 
-project = Project.from_file(Airplane.geometry, name="Python Project (Geometry, from file)")
+project = Project.from_cloud("prj-d3bcbd86-af18-43cc-8dc6-54ad1e19ce36")
```
Comment (maciej-flexcompute) [inline examples/tutorials/quick_start.py:23]: this will not work for everyone, right?

File: examples/tutorials/quick_start.py [hunk 2]
```diff
@@ -20,13 +20,38 @@
...
+    root_asset = project._root_asset
+
+    # Check if there are any new draft entities that have been added in the params by the user
+    entity_info = _set_up_param_entity_info(root_asset.entity_info, params)
+
+    with model_attribute_unlock(params.private_attribute_asset_cache, "project_entity_info"):
+        params.private_attribute_asset_cache.project_entity_info = entity_info
+    return params
+
+
```
Comment (maciej-flexcompute) [inline examples/tutorials/quick_start.py:54]: I don't think this should be part of quick start

File: flow360/component/simulation/outputs/output_entities.py [hunk 3]
```diff
@@ -41,6 +42,7 @@ class Slice(_SliceEntityBase):
     """Slice output item."""
 
     private_attribute_entity_type_name: Literal["Slice"] = pd.Field("Slice", frozen=True)
+    private_attribute_id: str = pd.Field(default=str(uuid.uuid4()), frozen=True)
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/outputs/output_entities.py:45]: I think every instance will have exactly the same uuid. Default factory should be used instead

File: flow360/component/simulation/services.py [hunk 4]
```diff
@@ -535,10 +535,16 @@ def generate_process_json(
...
+    params, errors, _ = validate_model(
         params_as_dict, unit_system_name, root_item_type, validation_level=validation_level
     )
 
+    if errors is not None:
+        raise ValueError(
+            "[Internal] Validation error occurred for supposedly validated param! Errors are: "
+            + str(errors)
+        )
+
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/services.py:547]: I like it! Often see "run draft error" with no info

File: flow360/component/simulation/services.py
Comment (maciej-flexcompute) [inline flow360/component/simulation/services.py:1]: should we remove line: 281:
```python
params = remove_properties_by_name(params, "_id")
```
?

File: examples/tutorials/quick_start.py (hunk 1 above)
Comment (benflexcompute) [reply examples/tutorials/quick_start.py:23]: Right, will remove.

File: flow360/component/simulation/services.py
Comment (benflexcompute) [reply flow360/component/simulation/services.py:1]: For backward compatibility maybe?

File: flow360/component/simulation/primitives.py [hunk 5]
```diff
@@ -259,10 +260,12 @@ def from_principal_axes(
         center: LengthType.Point,
         size: LengthType.PositiveVector,
         axes: OrthogonalAxes,
+        **kwargs,
     ):
         """
         Construct box from principal axes
         """
+        print("--- kwargs: ", kwargs)
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/primitives.py:268]: remove

**Review by maciej-flexcompute — APPROVED** [review]

File: flow360/component/simulation/primitives.py (hunk 5 above)
Comment (benflexcompute) [reply flow360/component/simulation/primitives.py]: Thanks for noticing it. Removed

**Pull request merged**